tests/test_client_api.py

```diff
@@ -32,9 +33,30 @@ def test_can_override_defaults(attr, value):
     assert attr in Session.__attrs__
     vanilla_session = Session()
 
-    client = APIClient("https://example.com", {attr: value})
+    client = APIClient("https://example.com", {attr: value})  # type: ignore[assignment]
 
     vanilla_value = getattr(vanilla_session, attr)
     client_value = getattr(client, attr)
     assert client_value != vanilla_value
     assert client_value == value
+
+
+def test_config_does_not_mutate():
+    imported_settings: RequestKwargs = {
+        "auth": ("djkhaled", "youplayedyourself"),
+    }
+
+    class MyConfig:
+        @staticmethod
+        def get_client_base_url():
+            return "https://example.com"
+
+        @staticmethod
+        def get_client_session_kwargs():
+            return imported_settings
+
+    one = APIClient.configure_from(MyConfig)
+    another_one = APIClient.configure_from(MyConfig)
+
+    assert one.auth == ("djkhaled", "youplayedyourself")
+    assert another_one.auth == ("djkhaled", "youplayedyourself")
```

Comment:
After reading #3... Is does this actually work? Or is this one of the surprises Victorien meant?

Reply:
it's a different thing. A surprising one indeed :grimacing: